tap_xero/client.py

```diff
@@ -8,10 +8,75 @@
 from singer.utils import strftime, strptime_to_utc
 import six
 import pytz
+import backoff
 
 BASE_URL = "https://api.xero.com/api.xro/2.0"
 
 
+class XeroError(Exception):
```

Comment:
Code added w.r.to TDL-648 (Permission Denied error should throw a user friendly message)